src/utils.jl

```diff
@@ -85,4 +85,12 @@ function proj2D(points::AbstractVector{Point{3,T}}) where {T}
   u = U[:,1]
   v = U[:,2]
   [Point(z⋅u, z⋅v) for z in eachcol(Z)]
-end
\ No newline at end of file
+end
+
+"""
+    dropunits(T)
```

Comment:
Thinking better, we need a different name because this dropunits function is acting on values and returning the type of the unitless type. It is not just dropping units. What about we introduce two helper functions dropunits(T) that takes a type and returns a type and utypeof(value) that takes a value and returns the unitless type? We could also take this opportunity and introduce a third function ueltype for the unitless eltype. What do you think? I think with these three helper functions we will be able to gradually fix implementations throughout the project.

Reply:
Also, it would be nice if we could add tests for these functions by adding Unitful.jl to the test dependencies if it is not already there. Maybe we should do this is in a separate PR and then rebase this PR here with the fix to the specific implementation.

Reply:
`dropunits`, `utypeof` and `ueltype` sound good, but they just sound. We're probably making simple things more complicated, after all, even without any helper functions we're only going to do what we can do in 14 characters. Wrapping something that short once is meant to be readable and easy to modify, while wrapping it 3 times somewhat complicates a simple problem. In addition, the u prefix actually implies unsigned, `ueltype` rather than `dropunits` brings more ambiguity. So, to sum up, I do not recommend this continued encapsulation of the modification.

Adding Unitful to the test is a big job. Rome was not built in a day. This package was not designed well at the beginning with the case with units in mind, for example even the definition of Ray is something that needs to be modified. We have to fix the current PR first, and then improve such a big project little by little. It is not practical to expect to fix them all in one PR.

Reply:
> `dropunits`, `utypeof` and `ueltype` sound good, but they just sound. We're probably making simple things more complicated, after all, even without any helper functions we're only going to do what we can do in 14 characters.

I don't understand how splitting different functions with different use cases complicates things, can you elaborate? You mean that we should stick to a single function name `dropunits` and write various methods for types, values, and collections of values? I think we can stick to `dropunits` for the first two use cases, but for vectors of values we need a separate function if that is actually necessary.

> Adding Unitful to the test is a big job. Rome was not built in a day. This package was not designed well at the beginning with the case with units in mind, for example even the definition of Ray is something that needs to be modified. We have to fix the current PR first, and then improve such a big project little by little. It is not practical to expect to fix them all in one PR.

I think you misunderstood my suggestion. The suggestion is to write a self-contained PR with the dropunits function and a few tests for this function, nothing more. Fixing units in the entire codebase in a single PR is not reasonable. After this dropunits function is in master with its functionality tested and documented, we can then come back to this PR here and use the dropunits function on it. Does it make sense?

Reply:
> I don't understand how splitting different functions with different use cases complicates things, can you elaborate? You mean that we should stick to a single function name dropunits and write various methods for types, values, and collections of values? I think we can stick to dropunits for the first two use cases, but for vectors of values we need a separate function if that is actually necessary.

We may have misunderstood each other's meaning. Let me rephrase my point.

First, it is true that the treatment of scalars and the treatment of vectors should be separated. Using a function that acts on both scalars and vectors can cause confusion.

In the case of scalars, we find that one and oneunit can be used on both values and types. Since Base has already done this, there is nothing wrong with designing dropunits that can be used on both types and values, just like Base. dropunits themselves do not need to be split up.

In the case of vectors, it is not possible to continue using dropunits. But we can nest and use a structure like dropunits(eltype(v)). It is not a bad idea to give this nested function another name. But ueltype is reminiscent of unsigned eltype, which is not a good name.

Reply:
> I think you misunderstood my suggestion. The suggestion is to write a self-contained PR with the dropunits function and a few tests for this function, nothing more. Fixing units in the entire codebase in a single PR is not reasonable. After this dropunits function is in master with its functionality tested and documented, we can then come back to this PR here and use the dropunits function on it. Does it make sense?

Here is where I misunderstood, I thought you wanted to fix the whole project in one PR. It's a good idea to open a PR just for adding these few functions.

Reply:
@HaoxuanGuo would you like to work on this PR for the dropunits utility function with tests? We can then come back to this PR later and rebase if necessary.

Reply:
@juliohm Of course. I would like to. Sorry for the PR closed due to my wrong operation.